src/flat.mjs

```diff
@@ -0,0 +1,29 @@
+import ensureIterable from './internal/ensure-iterable'
+
+function * flat (depth, iterable) {
+  if (depth === 0) {
+    yield * ensureIterable(iterable)
+  } else {
+    for (const iter of ensureIterable(iterable)) {
+      if (iter[Symbol.iterator]) {
```

Comment:
Should it be `if (typeof iter[Symbol.iterator] === 'function')`?

Reply:
This is based on the assumption that if you implement ```iter[Symbol.iterator]``` with anything different from a function or undefined, you are a bad person and deserve to spend an afternoon debugging. But I am going to fix it for sake of readability :smile:

Reply:
The whole check looks superfluous to me, it's already done in `ensureIterable` so we will always end up in the first branch here.

Reply:
I am not sure I follow.
ensureIterable ensures that the original iterable is an iterable. Then I check item by item if they are iterable or not.

Reply:
I'm sorry, I was reading wrong. The code is right.

Reply:
Better safe than sorry. Thanks for checking ;-)